cpp/test/random/sample_without_replacement.cu

```diff
@@ -214,7 +214,11 @@ const std::vector<SWoRInputs<float>> inputsf = {{1024, 512, -1, 0.f, GenPhilox,
         << "repeated index @i=" << i << " idx=" << val;                                            \
       occurrence.insert(val);                                                                      \
     }                                                                                              \
-    if (params.largeWeightIndex >= 0) { ASSERT_EQ(h_outIdx[0], params.largeWeightIndex); }         \
+    if (params.largeWeightIndex >= 0) {                                                            \
+      ASSERT_TRUE((h_outIdx[0] == params.largeWeightIndex) ||                                      \
+                  (h_outIdx[1] == params.largeWeightIndex) ||                                      \
+                  (h_outIdx[3] == params.largeWeightIndex));                                       \
```

Comment:
This looks like you're checking against the 1st, 2nd and 4th element's here - is this by design? 

Based off the PR description i would have expected this line to check the 3rd element instead of the 4th

Reply:
Good catch. I was checking first five elements first and the deleted wrong indices. Let me correct that. Thanks.